File: php/app/Models/Product.php
<?php

declare(strict_types=1);

namespace App\Models;

class Product
{
    public string $id;
    public string $name;
    public string $desc;
    public int $price;
    public string $user_id;
    public string $created_at;

    public function __construct(array $data)
    {
        $this->id         = $data['id'] ?? '';
        $this->name       = $data['name'] ?? '';
        $this->desc       = $data['desc'] ?? '';
        $this->price      = (int) ($data['price'] ?? 0);
        $this->user_id    = $data['user_id'] ?? '';
        $this->created_at = $data['created_at'] ?? '';
    }

    public function toArray(): array
    {
        return [
            'id'         => $this->id,
            'name'       => $this->name,
            'desc'       => $this->desc,
            'price'      => $this->price,
            'user_id'    => $this->user_id,
            'created_at' => $this->created_at,
        ];
    }
}

File: go/internal/models/product.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Product struct {
	ID        string    `gorm:"type:uuid;primaryKey" json:"id"`
	Name      string    `gorm:"not null" json:"name"`
	Desc      string    `json:"desc"`
	Price     int64     `gorm:"not null" json:"price"`
	UserID    string    `gorm:"not null" json:"user_id"`
	CreatedAt time.Time `json:"created_at"`
}

func (p *Product) BeforeCreate(tx *gorm.DB) error {
	if p.ID == "" {
		p.ID = uuid.NewString()
	}
	return nil
}
